### PR TITLE
Add Rusanov boundary correction to Newtonian Euler

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

Adds a Rusanov solver for Newtonian Euler, but implemented in the new boundary corrections framework.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

**Review by fmahebert — DISMISSED** [review]

File: src/Evolution/Systems/NewtonianEuler/BoundaryCorrections/Rusanov.hpp [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+ *
+ * Let \f$U\f$ be the state vector of evolved variables, \f$F^i\f$ the
+ * corresponding fluxes, and \f$n_i\f$ be the outward directed unit normal to
+ * the interface. Denoting \f$F := n_i F^i\f$, the %Rusanov boundary correction
+ * is
+ *
+ * \f{align*}
+ * G_\text{Rusanov} = \frac{F_\text{int} - F_\text{ext}}{2} -
+ * \frac{\text{max}\left(\{|\lambda_\text{int}|\},
+ * \{\lambda_\text{ext}|\}\right)}{2} \left(U_\text{ext} - U_\text{int}\right),
```
Comment (kidder) [inline src/Evolution/Systems/NewtonianEuler/BoundaryCorrections/Rusanov.hpp]: missing `|` before exterior char speed

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash fixing the typo

**Comment by nilsdeppe** [comment]
Rebased and squashed. Thanks for the reviews!

**Review by kidder — DISMISSED** [review]

**Comment by nilsdeppe** [comment]
I squashed in the `need_normal_vector` removal

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
@fmahebert this has been updated

**Review by fmahebert — APPROVED** [review]

**Pull request merged**